tell_me_your_secrets/__main__.py

```diff
@@ -130,35 +129,37 @@ def __init__(self, config_object: dict, search_path: str, use_gitignore: bool, p
         self.max_file_size = config_object.get('max_file_size', MAX_FILE_SIZE)
         self.write_results = write_results
         self.print_results = print_results
-        self.signatures = []
         self.matched_signatures = []
-        self.user_filters = user_filters
         self.output_path = output_path
         # $ Make Configuration Objects For each of the Signatures in the Config Object.
-        self.load_config()
+        self.signatures = self.load_signatures(config_object.get('signatures', {}), user_filters)
         module_logger.info(f'Secret Sniffer Initialised For Path: {search_path}')
 
     # $ Create the signature objects over here. 
-    def load_config(self):
+    @staticmethod
+    def load_signatures(raw_signatures: dict, user_filters: list) -> list:
         chosen_configs = []
-        for signature_obj in self.config['signatures']:
+        parsed_signatures = []
+        for signature_obj in raw_signatures:
             # $ Ignore Object if no Name/Part.
             if 'name' not in signature_obj or 'part' not in signature_obj:
                 continue
```

Comment:
It might be useful to log a warning here to inform the user that a signature in the config file was skipped due to missing name/part.